runelite-client/src/main/java/net/runelite/client/plugins/mining/MiningPlugin.java

```diff
@@ -123,13 +167,12 @@ public void onGameObjectDespawned(GameObjectDespawned event)
 
 		final GameObject object = event.getGameObject();
 		final int region = client.getLocalPlayer().getWorldLocation().getRegionID();
-
-		Rock rock = Rock.getRock(object.getId());
-		if (rock != null)
+		final Rock rock = Rock.getRock(object.getId());
+		if (rock == Rock.DENSE_RUNESTONE)
```

Comment:
with null-check not being required I meant only for runestones, but for addRockToRespawns it is still required.